core/observables/file.py

```diff
@@ -34,3 +36,39 @@ def info(self):
         i['hashes'] = self.hashes
         i['filenames'] = self.filenames
         return i
+
+
+class Certificate(Observable):
+
+    value = StringField(verbose_name="Value")
+    hashes = ListField(DictField(), verbose_name="Hashes")
+    body = ReferenceField("AttachedFile")
+
+    exclude_fields = Observable.exclude_fields + ['body']
+
+    @staticmethod
+    def check_type(txt):
+        return True
+
+    @classmethod
+    def from_data(cls, data, hash_sha256):
+        body = AttachedFile.from_content(
+            BytesIO(data), "cert.der", "application/pkix-cert",
+        )
+        return cls.get_or_create(
+            value="CERT:{}".format(hash_sha256), body=body,
+        )
+
+
+class CertificateSubject(Observable):
```

Comment:
Is this Observable Type really useful ? I think we could use `Text` instead.

Reply:
I though it was useful because `IvrePassive` acts on `CertificateSubject` observables, I don't know how to do this if I use a `Text`. Can you help me on that?

Reply:
I don't think you need to do anything but remove any reference to `CertificateSubject` and replace it with `Text`. This should work exactly the same.

Reply:
@tomchop that's it!

Reply:
Woops, typed to fast, missed the `Certificate` observable ^^. I understand @p-l- 's need for a `CertificateSubject` observable; here are some ideas:

* Maybe add a "subject" field to the `Certificate` observable (this may be redundant with another point below)
* Turn the `CertificateSubject` observable type into a `Text` observable, and extend it by adding a 'subtype' field (which I assume may come in handy in the future anyways).
* Have `IvrePassive` loop over text observables and `continue` if the subtype is not the expected one.

I think this solution is not great, especially in databases with lots of `Text` observables, since the db will end up seeking much more records than needed.

In any case, I see that currently the `Ip` observable is linked to both `Certificate` and `CertificateSubject`. I think it would be better to link the `Ip` to `Certificate` and `Certificate` to its `CertificateSubject`. Do you see any downsides to this?

Reply:
> I think this solution is not great, especially in databases with lots of `Text` observables, since the db will end up seeking much more records than needed.

Yep, that's why I have chosen another, specific observable. Also, the CertificateSubject can be the first observable, and may exist without a Certificate during an investigation (for example, if a certificate subject is shared as an IOC). 

> In any case, I see that currently the `Ip` observable is linked to both `Certificate` and `CertificateSubject`. I think it would be better to link the `Ip` to `Certificate` and `Certificate` to its `CertificateSubject`.

I think this is what actually happens. A docstring was saying otherwise, but I have double checked and fixed the docstring. Let me know if you think that's not the case.

Reply:
You're right @gaelmuller, it seems I can't GitHub today.

@p-l- , OneShot analytics will just analyze whatever observable you feed them through the interface. Scheduled analytics, on the other hand, are the ones that go through all the DB searching for stuff to analyze. So in this case it's perfectly safe to use a `Text` observable type (unless in the future you want to automate this analysis on all Certificate Subjects)

Reply:
Yep, that (going to scheduled analytics) is an option. Also, I don't think users would understand that they have an "IVRE - Passive" analytics they can apply to any Text observable, while it would only be useful for CertificateSubject. Is it important to limit the number of different observable types?

Reply:
It's probably best to be conservative as to how many different observable types we have, but I would say this justifies the creation of a new type.